src/adapter/sources.ts

```diff
@@ -275,16 +301,19 @@ export class SourceContainer {
   _fileContentOverridesForTest = new Map<string, string>();
 
   private _disabledSourceMaps = new Set<Source>();
-  private _scriptSkipper?: ScriptSkipper;
+  private _scriptSkipper: ScriptSkipper;
 
   constructor(
     dap: Dap.Api,
     private readonly sourceMapCache: SourceMapCache,
     public readonly rootPath: string | undefined,
     public readonly sourcePathResolver: ISourcePathResolver,
     public readonly localSourceMaps: ISourceMapRepository,
+    scriptSkipper: ScriptSkipper,
   ) {
     this._dap = dap;
+    this._scriptSkipper = scriptSkipper;
+    scriptSkipper.addNewSourceContainer(this);
```

Comment:
There is a circular dependency between ScriptSkipper and the SourceContainer which feels awkward but I'm not sure what to suggest

Reply:
Fixed!